flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsStateBackend.java

```diff
@@ -97,6 +100,27 @@ public FsStateBackend(String checkpointDataUri) throws IOException {
 	 *
 	 * @param checkpointDataUri The URI describing the filesystem (scheme and optionally authority),
 	 *                          and the path to the checkpoint data directory.
+	 * @param asynchronousSnapshots Switch to enable asynchronous snapshots.
+	 *
+	 * @throws IOException Thrown, if no file system can be found for the scheme in the URI.
+	 */
+	public FsStateBackend(String checkpointDataUri, boolean asynchronousSnapshots) throws IOException {
```

Comment:
We are getting one more parameter into the constructors with the change makes the state backend handle all checkpoint/savepoint storage related business. That must be constructor parameter, so if we can avoid further constructor parameters, that would help. Otherwise we really end up with 20 constructors.